src/ripple/app/paths/impl/DirectStep.cpp

```diff
@@ -603,6 +583,8 @@ TER DirectStepI::check (StrandContext const& ctx) const
             return ter;
     }
 
+    // Why do we only need to check this constraint if ctx.prevStep is a
+    // direct step?
```

Comment:
`NoRipple` is only relevant for account -> account -> account steps. See: https://ripple.com/build/understanding-the-noripple-flag/